pkg/where/match.go

```diff
@@ -144,10 +144,12 @@ func TreeGlob(field string, query string) string {
 func ConcatMatchKV(key, value string) string {
 	startLine := value[0] == '^'
 	endLine := value[len(value)-1] == '$'
-	if startLine {
+	if startLine && endLine {
 		return key + opEq + value[1:]
 	} else if endLine {
-		return key + opEq + value + "\\\\%"
+		return key + opEq + "\\\\%" + value
+	} else if startLine {
```

Comment:
We don't need else here, because if contains only return statement. (Also applies to the prevoius if clause)

```suggestion
	}
	if startLine {
```

Reply:
`else if` block require less lines and has better readability.